src/System.Net.Sockets/tests/FunctionalTests/UnixDomainSocketTest.cs

```diff
@@ -0,0 +1,221 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using System.Net.Test.Common;
+using System.Text;
+using System.Threading;
+
+using Xunit;
+using Xunit.Abstractions;
+
+namespace System.Net.Sockets.Tests
+{
+    public class UnixDomainSocketTest : IDisposable
+    {
+        private readonly ITestOutputHelper _log;
+        private readonly string _path;
+        private readonly UnixDomainSocketEndPoint _endPoint;
+
+        public UnixDomainSocketTest(ITestOutputHelper output)
+        {
+            _log = TestLogging.GetInstance();
+            _path = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+            if (File.Exists(_path))
+            {
+                File.Delete(_path);
```

Comment:
Wow, that is counter-intuitive. Should be a moot point in any case, since the path generation will presumably move out of the constructor in order to make the path selection contingent on a successful `Bind()`.